Robot-Tests/KW_Rautatien_huoltotie.robot

```diff
@@ -0,0 +1,276 @@
+
+    Click Element At Coordinates      ${Kartta}    200    0
+    Odota sivun latautuminen
+    Tupla Klikkaa Kartan Keskelle
+    Wait Until Element Is Visible     ${FA_otsikko}
+    Element Should Be Visible         ${FA_Lisätty_Järjestelmään}
+    Element Should Be Visible         ${FA_Muokattu_viimeksi}
+    Element Should Be Visible         ${Rautatien_huoltotie_tarkastelu_boolean}
+    Element Should Contain            ${Rautatien_huoltotie_tarkastelu_boolean}    on
+    Click Element At Coordinates      ${Kartta}    200    0
+    Odota sivun latautuminen
+
+
+Rautateiden huoltotien muokkaus koko ketjulle    [arguments]    ${testipaikka}    ${j}
+    Log    Muokataan rautatien huoltotietä koko ketjulle
+    Testin Aloitus
+    Vaihda Tietolaji    ${TL_Rautateiden_huoltotie_RB}
+    Paikanna osoite    ${testipaikka}
+    Zoomaa kartta    2    200
+    Odota sivun latautuminen
+
+    Click Element At Coordinates      ${Kartta}    0    20
+    Wait Until Element Is Visible     ${FA_otsikko}
+    Element Should Be Visible         ${FA_Lisätty_Järjestelmään}
+    Element Should Be Visible         ${FA_Muokattu_viimeksi}
+    Element Should Be Visible         ${Rautatien_huoltotie_tarkastelu_boolean}
+    Element Should Contain            ${Rautatien_huoltotie_tarkastelu_boolean}    on
+    Siirry Muokkaustilaan
+
+    Select Radio Button               maintenanceRoad   enabled
+    Radio Button Should Be Set To     maintenanceRoad   enabled
+    Click Element    ${huoltotie_käyttöoikeus_pudotusvalikko}
+
+    FOR    ${j}    IN    @{Huoltotie_käyttöoikeus_kohteet}
+        Log    ${j}
+        Press Keys    None    ARROW_DOWN
+        Element Should Contain    ${huoltotie_käyttöoikeus_pudotusvalikko}    ${j}
+        Press Keys    None    ENTER
+    END
+
+    Element Should Be Enabled    ${FA_footer_Peruuta}
+    Click Element                ${FA_footer_Tallenna}
+
+
+Rautateiden huoltotien muokkaus ketjun osalle    [arguments]    ${testipaikka}    ${k}
+    Log    Muokataan rautatien huoltotietä ketjun osalle
+    Testin Aloitus
+    Vaihda Tietolaji    ${TL_Rautateiden_huoltotie_RB}
+    Paikanna osoite    ${testipaikka}
+    Zoomaa kartta    2    200
+    Odota sivun latautuminen
+
+    Click Element At Coordinates      ${Kartta}    0    20
+    Wait Until Element Is Visible     ${FA_otsikko}
+    Element Should Be Visible         ${FA_Lisätty_Järjestelmään}
+    Element Should Be Visible         ${FA_Muokattu_viimeksi}
+    Element Should Be Visible         ${Rautatien_huoltotie_tarkastelu_boolean}
+    Element Should Contain            ${Rautatien_huoltotie_tarkastelu_boolean}    on
+    Siirry Muokkaustilaan
+
+    Select Radio Button               maintenanceRoad   enabled
+    Radio Button Should Be Set To     maintenanceRoad   enabled
+    Click Element    ${huoltotie_käyttöoikeus_pudotusvalikko}
+    Click Element    ${huoltotie_käyttöoikeus_pudotusvalikko}
+
+    FOR    ${k}    IN    @{Huoltotie_käyttöoikeus_kohteet}
```

Comment:
Onko tässä sama loop kahteen kertaan, kun test casessakin on tämä?

Reply:
Keywordien puolella oleva loop koskee pudotusvalikon manipulointia. Tavallaan "nice to have" -ominaisuus.